kpet/targeted.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2018 Red Hat, Inc. All rights reserved. This copyrighted
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General Public
+# License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+"""Module where targeted testing is implemented"""
+import os
+import json
+import re
+
+
+class UnrecognizedPatchFormat(Exception):
+    """Raised when can't extract any source file from a patch"""
+
+
+def get_patterns_by_layout(layout, dbdir):
+    """Return patterns of layout"""
```

Comment:
Please document the parameters *and* the return values, here and in all other functions.
Check skt source code for examples.

I would also put `dbdir` in front, as it's more fundamental, but would leave that up to you.

Reply:
Done, but I didn't swap the parameters :man_shrugging: